src/utils.js

```diff
@@ -150,6 +152,21 @@ function altContigName(contig: string): string {
   }
 }
 
+/**
+ * Pipe all promise events through a deferred object.
+ * This is similar to deferred.resolve(promise), except that it allows progress
```

Comment:
I am definitely not a promises expert, but this seems different than `.resolve()`? It's just adding a `then` to the deferred?